coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CoralToTrinoSqlCallConverter.java

```diff
@@ -0,0 +1,169 @@
+        "[{\"regex\":\"(?i)('utf-8')\", \"input\":2, \"name\":\"from_utf8\"}]", "[{\"input\":1}]", null, null));
+    sqlCallTransformerList.add(new OperatorBasedSqlCallTransformer(hiveToCoralSqlOperator("date_add"), 2, "date_add",
+        "[{\"value\": 'day'}, {\"input\": 2},  "
+            + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+        null, null));
+    sqlCallTransformerList.add(new OperatorBasedSqlCallTransformer(hiveToCoralSqlOperator("date_sub"), 2, "date_add",
+        "[{\"value\": 'day'}, " + "{\"op\": \"*\", \"operands\":[{\"input\": 2}, {\"value\": -1}]}, "
+            + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+        null, null));
+    sqlCallTransformerList.add(new OperatorBasedSqlCallTransformer(hiveToCoralSqlOperator("datediff"), 2, "date_diff",
+        "[{\"value\": 'day'}, {\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 2}]}]}, "
+            + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+        null, null));
+    sqlCallTransformerList.add(new ToDateOperatorTransformer(configs.getOrDefault(AVOID_TRANSFORM_TO_DATE_UDF, false)));
+    sqlCallTransformerList.add(new MapStructAccessOperatorTransformer());
+
+    // LinkedIn specific functions
+    addLinkedInFunctionTransformer("com.linkedin.dali.udf.watbotcrawlerlookup.hive.WATBotCrawlerLookup", 3,
+        "wat_bot_crawler_lookup");
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.parsing.hive.Ip2Str", 1, "ip2str");
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.parsing.hive.Ip2Str", 3, "ip2str");
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.parsing.hive.UserAgentParser", 2, "useragentparser");
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.lookup.hive.BrowserLookup", 3, "browserlookup");
+    addLinkedInFunctionTransformer("com.linkedin.jobs.udf.hive.ConvertIndustryCode", 1, "converttoindustryv1");
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.urnextractor.hive.UrnExtractorFunctionWrapper", 1,
+        "urn_extractor");
+    addLinkedInFunctionTransformer("com.linkedin.stdudfs.hive.daliudfs.UrnExtractorFunctionWrapper", 1,
+        "urn_extractor");
+    addLinkedInFunctionTransformerFromHiveRegistry();
+
+    this.sqlCallTransformers = SqlCallTransformers.of(ImmutableList.copyOf(sqlCallTransformerList));
+  }
+
+  private void addLinkedInFunctionTransformer(String linkedInFuncName, int numOperands, String trinoFuncName) {
+    this.sqlCallTransformerList
+        .add(new LinkedInOperatorBasedSqlCallTransformer(linkedInFuncName, numOperands, trinoFuncName));
+    this.linkedInFunctionSignatureSet.add(linkedInFuncName.toLowerCase() + "_" + numOperands);
+  }
+
+  private void addLinkedInFunctionTransformerFromHiveRegistry() {
+    ImmutableMultimap<String, Function> registry = HIVE_FUNCTION_REGISTRY.getRegistry();
```

Comment:
I do not think we need to iterate over the registry since we already have the function name coming in from the `condition` and `visit`. Thoughts @ljfgem?

Reply:
Agreed, `GenericCoralRegistryOperatorRenameSqlIdentifierTransformer` is much cleaner.